### PR TITLE
Automating needs triage for new issues

### PR DESCRIPTION
**Pull request by Rumyra**
#### Summary

Adding a workflow to add the `needs-triage` label automatically to new issues

#### Supporting details

Followed https://docs.github.com/en/actions/managing-issues-and-pull-requests/adding-labels-to-issues

#### Metadata

- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

### CODE REVIEW

File: .github/workflows/new-issues.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: 'Mark new issues with needs-triage label'
+
+on:
+  issues:
+    types:
+      - reopened
+      - opened
+
+jobs:
+  label-new-issues
```
Comment (schalkneethling) [inline .github/workflows/new-issues.yml:10]: I believe this should be:

```suggestion
  label-new-issues:
```

File: .github/workflows/new-issues.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    runs-on: ubuntu-latest
+    permissions:
+      issues: write
+    steps:
+      - name: initial labelling
+        uses: andymckay/labeler@master
+        with:
+          add-labels: "needs-triage"
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (schalkneethling) [inline .github/workflows/new-issues.yml:19]: I do not see the need for this token in any of the examples in https://github.com/andymckay/labeler
Perhaps the `GITHUB_TOKEN` that is provided by default by GitHub is good enough for this action purpose?

File: .github/workflows/new-issues.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  issues:
+    types:
+      - reopened
+      - opened
+
+jobs:
+  label-new-issues
+    runs-on: ubuntu-latest
+    permissions:
+      issues: write
```
Comment (schalkneethling) [inline .github/workflows/new-issues.yml:13]: I also do not see any of the examples specifying any permission so, perhaps it is not needed?

**Review by schalkneethling — CHANGES_REQUESTED** [review]

File: .github/workflows/new-issues.yml (hunk 3 above)
Comment (Rumyra) [reply .github/workflows/new-issues.yml:13]: This has a permissions in the example https://docs.github.com/en/actions/managing-issues-and-pull-requests/adding-labels-to-issues 🤷🏻‍♀️

Should I remove it and try it and if it doesn't work add it back?

File: .github/workflows/new-issues.yml (hunk 2 above)
Comment (Rumyra) [reply .github/workflows/new-issues.yml:19]: This example https://docs.github.com/en/actions/managing-issues-and-pull-requests/adding-labels-to-issues - same as below? Remove, try it and see...

**Pull request merged**